### PR TITLE
windows11-fonts: init at 10.0.22631.2428-2, windows10-fonts: init at 10.0.19045.2006-1

### PR DESCRIPTION
**Pull request by Sobte**
## Description of changes

I want to add windows fonts, just like [vista-fonts](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/vista-fonts/default.nix) and [ttf-ms-win11-auto](https://aur.archlinux.org/packages/ttf-ms-win11-auto)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://[23.05](https://ggithub.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/fonts/windows-fonts/default.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  p7zip,
+  # To select only certain languages, put a list of strings to `languages`: every key in
+  # ./${pname}-languages.nix is an optional language
+  languages ? [ ],
```
Comment (Cryolitia) [inline pkgs/data/fonts/windows-fonts/default.nix:8]: It's more recommand to use `outputs`, see: https://ryantm.github.io/nixpkgs/stdenv/multiple-output/

File: pkgs/data/fonts/windows-fonts/default.nix [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        install -Dm644 ${fontPaths} -t $out/share/fonts/opentype
+        install -Dm644 Windows/System32/Licenses/neutral/*/*/license.rtf -t $out/share/licenses
+
+        runHook postInstall
+      '';
+
+      meta = {
+        description = "Some TrueType fonts from ${desc} (Arial, Bahnschrift, Calibri, Cambria, Candara, Consolas, Constantia...)";
+        homepage = "https://learn.microsoft.com/typography/";
+        license = lib.licenses.unfree;
```
Comment (Cryolitia) [inline pkgs/data/fonts/windows-fonts/default.nix:61]: Is there any EULA?

File: pkgs/data/fonts/windows-fonts/default.nix [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        # fontconfig configuration files.
+        priority = 5;
+      };
+    };
+in
+{
+  windows11 = windows-fonts {
+    pname = "windows11-fonts";
+    version = "10.0.22631.2428-2";
+    url = "https://software-static.download.prss.microsoft.com/dbazure/888969d5-f34g-4e03-ac9d-1f9786c66749/22631.2428.231001-0608.23H2_NI_RELEASE_SVC_REFRESH_CLIENTENTERPRISEEVAL_OEMRET_x64FRE_en-us.iso";
```
Comment (Cryolitia) [inline pkgs/data/fonts/windows-fonts/default.nix:74]: Download a whole ISO may put a heavy burden on hydra

**Comment by drupol** [comment]
It looks like you accidentally mass-pinged a bunch of people, which are now subscribed
and getting notifications for everything in this pull request. Unfortunately, they
cannot be automatically unsubscribed from the issue (removing review request does not
unsubscribe), therefore development cannot continue in this pull request anymore.

Please create a new pull request, link back to this one and ping the
people actually involved in here over there. For the next time, remember to set your PR to draft status before rebasing. In draft status, you can preview the list of maintainers that are about to be requested for review, which allows you to sidestep this issue.

In order to avoid this in the future, there are instructions for how to properly
rebase between branches in our [contribution guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging).
Setting your pull request to draft prior to rebasing is strongly recommended.
In draft status, you can preview the list of people that are about to be requested
for review, which allows you to sidestep this issue.
This is not a bulletproof method, though, as OfBorg still does review requests even on draft PRs.

**Pull request closed without merging**